openslides/agenda/static/js/agenda/site.js

```diff
@@ -134,8 +191,19 @@ angular.module('OpenSlidesApp.agenda.site', ['OpenSlidesApp.agenda'])
         // delete selected item
         $scope.delete = function () {
             angular.forEach($scope.items, function (item) {
-                if (item.selected)
-                    Agenda.destroy(item.id);
+                if (item.selected) {
+                    switch (item.content_object.collection) {
+                        case 'core/customslide':
+                            Customslide.destroy(item.content_object.id);
+                            break;
+                        case 'motions/motion':
+                            Motion.destroy(item.content_object.id);
+                            break;
+                        case 'assignments/assignment':
+                            Assignment.destroy(item.content_object.id);
+                            break;
+                    }
+                }
```

Comment:
I don't like this code, because the agenda app should be independent from Motion and Assignment. But it is not needed anyway, because the Customslide-, Motion- or Assignmentobject is deleted from the server when the related item is deleted.